netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -365,20 +366,16 @@ void overrideAuthorityChecker(@Nullable OverrideAuthorityChecker authorityChecke
   @VisibleForTesting
   @CheckReturnValue
   static ProtocolNegotiator createProtocolNegotiator(
+      SocketAddress targetServerAddress,
       String authority,
       NegotiationType negotiationType,
-      SslContext sslContext) {
+      SslContext sslContext
+  ) {
     ProtocolNegotiator negotiator =
         createProtocolNegotiatorByType(authority, negotiationType, sslContext);
-    String proxy = System.getenv("GRPC_PROXY_EXP");
+    SocketAddress proxy = Proxies.proxyFor(targetServerAddress);
```

Comment:
This makes me worried for tests, since this uses ProxySelector with its global auto-configuration.